internal/content_coding.go

```diff
@@ -72,12 +73,14 @@ func (r *GzipReader) Read(b []byte) (int, error) {
 }
 
 // NewContentEncoder returns a ContentEncoder for the encoding type.
-func NewContentEncoder(encoding string) (ContentEncoder, error) {
+func NewContentEncoder(encoding string, level int) (ContentEncoder, error) {
```

Comment:
Not a must, just an idea: Should we adopt the [functional option pattern](https://golang.cafe/blog/golang-functional-options-pattern.html) here? This would allow to set a default level and keep the API stable to what we had before. At the same time it allows to extend the encoders later with more options...

Reply:
I wanted something like that. I'll play with it more.